packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -2321,6 +2346,16 @@ function commitMutationEffectsOnFiber(
       }
       break;
     }
+    case Fragment:
+      if (enableFragmentRefs) {
+        if (current && current.stateNode !== null) {
+          updateFragmentInstanceFiber(finishedWork, current.stateNode);
+        }
+        if (flags & Ref) {
+          safelyAttachRef(finishedWork, finishedWork.return);
```

Comment:
I don't know where this pattern is coming from but this should be in `reappearLayoutEffects`, not here. Attaching refs happen in the layout effects which can happen either in `commitLayoutEffects` or `reappearLayoutEffects`.

Reply:
I think it's not safe to check the flag because it would've already been reset at this point. The reappearing doesn't always happen on a tree that was just rendered so those can be old Fibers. I think it would have to be like a static flag. I think the TODO isn't quite right (which isn't unusual for TODOs).